lib/Adapter/WorseReflection/Refactor/WorseGenerateAccessor.php

```diff
@@ -112,4 +105,41 @@ private function sourceFromSymbolInformation(SourceCode $sourceCode, SymbolConte
             $worseSourceCode->path()
         );
     }
+
+    /**
+     * Finds the class containing the property.
+     *
+     * @param string $source The source code of the file.
+     * @param int $offset The position where the cursor was, used to identify the correct
+     * class if there is more than one in the `$source`.
+     *
+     * @return ReflectionClass
+     *
+     * @throws InvalidArgumentException If there is no class in the code.
+     * @throws RuntimeException If it's impossible to determine which class to use.
+     */
+    private function class(string $source, int $offset): ReflectionClass
+    {
+        $classes = $this->reflector->reflectClassesIn($source);
+
+        if (0 === $classes->count()) {
+            throw new InvalidArgumentException(
+                'No classes in source file'
+            );
+        }
+
+        if (1 === $classes->count()) {
+            return $classes->first();
+        }
+
+        foreach ($this->reflector->reflectClassesIn($source) as $class) {
+            $position = $class->position();
+
+            if ($position->start() <= $offset && $offset <= $position->end()) {
+                return $class;
+            }
+        }
```

Comment:
_If_ we were to add a helper class (and interface?) in this library, it could be shared with the RPC extension (which already depends on this package).

Reply:
It would indeed be better.
Work for another time or do you want it with this PR ?

Reply:
I think it would be fine to create an issue and deal with it later. In general I have noticed quite a number of patterns which are repeated, so would be good to isolate them.